From f1355dfcd6cf0a4d413d7b04c2c166b825d559ce Mon Sep 17 00:00:00 2001
From: Diego Biurrun <diego@biurrun.de>
Date: Wed, 11 Jan 2012 10:50:35 +0100
Subject: [PATCH 03/63] doc: Replace ffmpeg references in avserver config file
 by avconv.

---
 doc/avserver.conf |   12 ++++++------
 1 files changed, 6 insertions(+), 6 deletions(-)

diff --git a/doc/avserver.conf b/doc/avserver.conf
index 2105ac5..1f21f4d 100644
--- a/doc/avserver.conf
+++ b/doc/avserver.conf
@@ -31,16 +31,16 @@ NoDaemon
 
 ##################################################################
 # Definition of the live feeds. Each live feed contains one video
-# and/or audio sequence coming from an ffmpeg encoder or another
+# and/or audio sequence coming from an avconv encoder or another
 # avserver. This sequence may be encoded simultaneously with several
 # codecs at several resolutions.
 
 <Feed feed1.ffm>
 
-# You must use 'ffmpeg' to send a live feed to avserver. In this
+# You must use 'avconv' to send a live feed to avserver. In this
 # example, you can type:
 #
-# ffmpeg http://localhost:8090/feed1.ffm
+# avconv http://localhost:8090/feed1.ffm
 
 # avserver can also do time shifting. It means that it can stream any
 # previously recorded live stream. The request should contain:
@@ -55,10 +55,10 @@ FileMaxSize 200K
 # ReadOnlyFile /saved/specialvideo.ffm
 # This marks the file as readonly and it will not be deleted or updated.
 
-# Specify launch in order to start ffmpeg automatically.
-# First ffmpeg must be defined with an appropriate path if needed,
+# Specify launch in order to start avconv automatically.
+# First avconv must be defined with an appropriate path if needed,
 # after that options can follow, but avoid adding the http:// field
-#Launch ffmpeg
+#Launch avconv
 
 # Only allow connections from localhost to the feed.
 ACL allow 127.0.0.1
-- 
1.7.5.4

