From e01a29a461ec15aad745fa89e510a92918d872aa Mon Sep 17 00:00:00 2001
From: Anton Khirnov <anton@khirnov.net>
Date: Thu, 12 Jan 2012 09:33:16 +0100
Subject: [PATCH 21/67] avserver: remove code using deprecated
 CODEC_CAP_PARSE_ONLY.

---
 avserver.c |   20 --------------------
 1 files changed, 0 insertions(+), 20 deletions(-)

diff --git a/avserver.c b/avserver.c
index 9016d1f..15fa64c 100644
--- a/avserver.c
+++ b/avserver.c
@@ -2112,22 +2112,6 @@ static void compute_status(HTTPContext *c)
     c->buffer_end = c->pb_buffer + len;
 }
 
-/* check if the parser needs to be opened for stream i */
-static void open_parser(AVFormatContext *s, int i)
-{
-    AVStream *st = s->streams[i];
-    AVCodec *codec;
-
-    if (!st->codec->codec) {
-        codec = avcodec_find_decoder(st->codec->codec_id);
-        if (codec && (codec->capabilities & CODEC_CAP_PARSE_ONLY)) {
-            st->codec->parse_only = 1;
-            if (avcodec_open2(st->codec, codec, NULL) < 0)
-                st->codec->parse_only = 0;
-        }
-    }
-}
-
 static int open_input_stream(HTTPContext *c, const char *info)
 {
     char buf[128];
@@ -2173,10 +2157,6 @@ static int open_input_stream(HTTPContext *c, const char *info)
         return -1;
     }
 
-    /* open each parser */
-    for(i=0;i<s->nb_streams;i++)
-        open_parser(s, i);
-
     /* choose stream as clock source (we favorize video stream if
        present) for packet sending */
     c->pts_stream_index = 0;
-- 
1.7.5.4

