From 4ad56612f94f5e4312ba24155b5c446e151e6ca1 Mon Sep 17 00:00:00 2001
From: Diego Biurrun <diego@biurrun.de>
Date: Thu, 23 Jun 2011 12:38:03 +0200
Subject: [PATCH 17/57] build: Remove dependency and editor backup files also in the doc/ subdirectory.

---
 Makefile |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/Makefile b/Makefile
index 084e175..d6bae76 100644
--- a/Makefile
+++ b/Makefile
@@ -172,7 +172,7 @@ testclean:
 clean:: testclean
 	$(RM) $(ALLPROGS)
 	$(RM) $(CLEANSUFFIXES)
-	$(RM) doc/*.html doc/*.pod doc/*.1
+	$(RM) doc/*.html doc/*.pod doc/*.1 doc/*.d doc/*~
 	$(RM) $(TOOLS)
 	$(RM) $(CLEANSUFFIXES:%=tools/%)
 
-- 
1.7.4.1

