From 924b2ee8f29823877b6b544b48fe153f307d7be7 Mon Sep 17 00:00:00 2001
From: Reinhard Tartler <siretart@tauware.de>
Date: Sun, 3 Jul 2011 20:00:31 +0200
Subject: [PATCH 37/57] Add version number to doxygen config

---
 Doxyfile |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/Doxyfile b/Doxyfile
index a4beaba..485b93d 100644
--- a/Doxyfile
+++ b/Doxyfile
@@ -31,7 +31,7 @@ PROJECT_NAME           = Libav
 # This could be handy for archiving the generated documentation or
 # if some version control system is used.
 
-PROJECT_NUMBER         =
+PROJECT_NUMBER         = 0.7.1
 
 # The OUTPUT_DIRECTORY tag is used to specify the (relative or absolute)
 # base path where the generated documentation will be put.
-- 
1.7.4.1

