From d32b2d4de164d6e2bd4b441c5f2aa0e442fef1f6 Mon Sep 17 00:00:00 2001
From: Reinhard Tartler <siretart@tauware.de>
Date: Sun, 3 Jul 2011 19:54:05 +0200
Subject: [PATCH 38/57] update Changelog

---
 Changelog |    3 +++
 1 files changed, 3 insertions(+), 0 deletions(-)

diff --git a/Changelog b/Changelog
index 276812b..1c6952c 100644
--- a/Changelog
+++ b/Changelog
@@ -7,6 +7,9 @@ version 0.7.1:
 - added various additional FOURCC codec identifiers
 - H.264 4:4:4 fixes
 - build system fixes
+- Doxygen corrections and improvements
+- fixed segfault in ffprobe
+- behavioral fix in av_open_input_stream()
 
 
 version 0.7:
-- 
1.7.4.1

