--- a/version.sh
+++ b/version.sh
@@ -6,6 +6,12 @@ test $revision || revision=`cd "$1" && g
 test $revision || revision=`cd "$1" && sed -n -e '/^dir$/{n;p;q}' .svn/entries 2>/dev/null`
 test $revision && revision=SVN-r$revision
 
+# check for debian package branch
+if test -f "$1"/debian/control; then
+    debrevision=`cd "$1" && dpkg-parsechangelog | awk '/^Version/ {print $2}'`
+    revision="r11872+debian_${debrevision}"
+fi
+
 # check for git short hash
 if ! test $revision; then
     revision=`cd "$1" && git log -1 --pretty=format:%h`
