Author: Reinhard Tartler <siretart@ubuntu.com>
Description: filter name needs to be double 0 terminated
Forwarded: http://bugzilla.libav.org/show_bug.cgi?id=35
Bug-Gnome: https://bugzilla.gnome.org/show_bug.cgi?id=654634

--- a/libpostproc/postprocess.c
+++ b/libpostproc/postprocess.c
@@ -763,7 +763,8 @@ pp_mode *pp_get_mode_by_name_and_quality
     ppMode->maxClippedThreshold= 0.01;
     ppMode->error=0;
 
-    av_strlcpy(temp, name, GET_MODE_BUFFER_SIZE);
+    memset(temp, 0, GET_MODE_BUFFER_SIZE);
+    av_strlcpy(temp, name, GET_MODE_BUFFER_SIZE - 1);
 
     av_log(NULL, AV_LOG_DEBUG, "pp: %s\n", name);
 
@@ -819,7 +820,7 @@ pp_mode *pp_get_mode_by_name_and_quality
 
                 plen= strlen(p);
                 spaceLeft= p - temp + plen;
-                if(spaceLeft + newlen  >= GET_MODE_BUFFER_SIZE){
+                if(spaceLeft + newlen  >= GET_MODE_BUFFER_SIZE - 1){
                     ppMode->error++;
                     break;
                 }
