Fix build failures for the GNU Hurd OS
===================================================================
Index: ffmpeg-debian/configure
===================================================================
--- ffmpeg-debian.orig/configure	2009-07-06 13:13:43.000000000 -0400
+++ ffmpeg-debian/configure	2009-07-06 13:13:49.000000000 -0400
@@ -1651,6 +1651,9 @@
         ;;
     gnu/kfreebsd)
         ;;
+    gnu)
+        disable dv1394
+        ;;
 
     *)
         die "Unknown OS '$target_os'."
